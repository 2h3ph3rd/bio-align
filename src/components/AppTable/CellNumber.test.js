import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import CellNumber from './CellNumber'

const fakeCell = {
  value: 1,
  top: true,
  left: false,
  diag: true,
}

it('renders correctly', () => {
  const tree = renderer
    .create(<CellNumber cell={fakeCell}></CellNumber>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render the letter given', () => {
  render(<CellNumber cell={fakeCell}></CellNumber>)

  const div = screen.getByText(fakeCell.value)
  expect(div).toHaveTextContent(fakeCell.value)
})
