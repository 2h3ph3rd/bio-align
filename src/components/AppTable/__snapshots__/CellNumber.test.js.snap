// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rhjugb-MuiPaper-root"
>
  <div
    className="MuiBox-root css-1nvf2ph"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ro2m3g-MuiSvgIcon-root"
      data-testid="ArrowUpwardIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-f9jn21-MuiSvgIcon-root"
      data-testid="ArrowUpwardIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wvslln-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <p
      className="MuiTypography-root MuiTypography-body1 css-rrlk9y-MuiTypography-root"
    >
      1
    </p>
  </div>
</div>
`;
